plotly/tools.py

```diff
@@ -1670,6 +1722,321 @@ def create_streamline(x, y, u, v,
                              mode='lines', **kwargs)
         return streamline
 
+    @staticmethod
+    def create_ohlc(open, high, low, close, direction, **kwargs):
+
+        """
+        Returns data for an ohlc chart
+
+        :param (list) open: opening values
+        :param (list) high: high values
+        :param (list) low: low values
+        :param (list) close: closing values
+        :param (string) direction: direction can be 'increasing' or
+            'decreasing'. When the direction is 'increasing', the returned data
+            consists of all units where the close value is greater than the
+            corresponding open value, and when the direction is 'decreasing',
+            the returned data consists of all units where the close value is
+            less than or equal to the corresponding open value.
+        :param (class) kwargs: kwargs passed through plotly.graph_objs.Scatter
+            for more information on valid kwargs call
+            help(plotly.graph_objs.Scatter)
+
+        :rtype (trace): returns data for ohlc increasing units or decreasing
+            units.
+
+        Example 1: Plot ohlc chart
+        ```
+        import plotly.plotly as py
+        import plotly.tools as tls
+        from plotly.graph_objs import *
+
+        # Add data
+        open_data = [33.0, 33.3, 33.5, 33.0, 34.1]
+        high_data = [33.1, 33.3, 33.6, 33.2, 34.8]
+        low_data = [32.7, 32.7, 32.8, 32.6, 32.8]
+        close_data = [33.0, 32.9, 33.3, 33.1, 33.1]
+
+        # Create ohlc increasing units
+        ohlc_increase = tls.TraceFactory.create_ohlc(open_data, high_data,
+                                                     low_data, close_data,
+                                                     direction='increasing')
+
+        # Create ohlc decreasing units
+        ohlc_decrease = tls.TraceFactory.create_ohlc(open_data, high_data,
+                                                     low_data, close_data,
+                                                     direction='decreasing')
```

Comment:
yeah, whoops prematuraly deleted my comment. Options:

```
traces = create_ohlc(..., direction='both') # | 'increasing' | 'decreasing'

traces[0].update({'line': {'color': 'blue'}})
traces[1].update({'line': {'color': 'red'}})
```

Other options:
1 - Let user pass through `Scatter` `**kwargs` that apply to both of the traces? e.g. `create_ohlc(..., line=Line(width=4), name='AAPL')`? Not sure. Seems sort of out of line of our declarative philosophy.
2 - Let user apply `kwargs` values as an array, e.g. `create_ohlc(...name=['AAPL Increasing', 'AAPL Decreasing'])`. I vote No: too confusing, too much to teach.

The most common attribute that people will want to edit will be the `name`. I think by default we should `group` the traces and hide one of the legend entries as I imagine most finance folks reading these charts won't need to see the `increasing` and `decreasing` legend labels, but might want to see the tickers (especially if multiple tickers are graphed) e.g. `AAPL`, `COKE`. With grouped and half-hidden legend entries, they'll just have to:
```
traces = create_ohlc(...)
traces[0].update({'name': 'AAPL'})
```